File: 240924/인접하지 않은 2개의 숫자/two-non-adjacent-numbers.cpp
#include <iostream>
using namespace std;
int main() {
    int n,max=0;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++)
        cin>>arr[i];
    for(int i=0;i<n-2;i++)
    {
        for(int j=i+2;j<n;j++)
        {
            if(arr[i] + arr[j] > max) max = arr[i] + arr[j];
        }
    }
    cout<<max;
    return 0;
}

File: 241016/인접하지 않은 2개의 숫자/two-non-adjacent-numbers.py
import sys

INT_MIN = - sys.maxsize
n = int(input())

arr = list(map(int, input().split()))

answer = INT_MIN
for i in range(n):
    sum = 0
    for j in range(i+2,n):
        sum = arr[i] + arr[j]
    answer = max(answer, sum)
print(answer)